eureka2-server/src/main/java/com/netflix/eureka2/server/service/SelfRegistrationService.java

```diff
@@ -77,8 +77,10 @@ public void call() {
                         public void call(Throwable throwable) {
                             logger.warn("Error during self unregistration process: {}", latestSelfInfo);
                         }
-                    }).materialize().toBlocking().firstOrDefault(null);
+                    }).materialize().toBlocking().firstOrDefault(null);  // FIXME why materialize and block here?
```

Comment:
I see. But shouldn't we let unregisters be best effort? Server side we are ok with best effort unregisters, and we would not want a bad unregister to block server shutdown.